### ISSUE TITLE
Select with Ajax fetch!

### ISSUE THREAD

**Issue by marcosn41** [wearebraid/vue-formulate#299]
I am using the formulate in the new version of my saas and I find it amazing how easy it is to work. But I am a new user on vue and I think one thing that is missing is a select with ajax. I am trying to put the vue-select component as a custom input but still without success. It would be cool to have that!

**Comment by justin-schroeder** [comment]
2 Ideas here:

1. It should be pretty easy to wrap vue-select as a [custom input](https://vueformulate.com/guide/inputs/custom-inputs/). I've never done this, but it seems reasonable.
2. What feature of vue-select are you trying to use? If you're just wanting to load your options from ajax this is pretty easy to do already. Here's a super simple example I whipped up: https://codepen.io/justin-schroeder/pen/gOMmeGz

**Comment by marcosn41** [comment]
Hey Thanks for answer!


I need him to have a search that looks for the backend, I'm trying to convert the example of the custom input from the autocomplete, where he fetches the server with the term entered. I think his operation fits what I need. but basically it is when typing it to search the database. What I haven't thought about yet is how to separate the label's value since it's an input and not a select.

Follow my evolution, suggestions are welcome too lol.

Anyway, it would be really nice to have that straight in the form, I think today is a common field.

**Comment by marcosn41** [comment]
Schema input call

`{ component: 'div', class: 'col-12', children: [{
            type: 'selectajax',
            name: 'idresponsavel',
            label: 'Responsável',
            endpoint: '/api/s/pessoas/'
          }]
        }`

endpoint its url for fetch, below is custom input watch code:

`watch: {
    model () {
      this.selectedIndex = 0
    }
    ,
    filteredOptions: {
      deep: true,
      handler: function(){
        this.context.options = this.getData(this.context.endpoint, this.context.model);
      }
    }
  }`

and getData function

`async getData(endpoint, vsearch){
      let cendpoint = `/api/s/pessoas/?search=${escape(vsearch)}`;
      let data = await apiService(cendpoint)
      return data.results;
    }`

About vue-select, i`m trying to do exactly like this:

[vue-select ajax](https://vue-select.org/guide/ajax.html)

**Comment by marcosn41** [comment]
Ok! I make the first option work! Is very easy to make custom input! Soon i`ll post like a plugin. Thanks

**Comment by justin-schroeder** [comment]
Awesome @marcosn41 would love to see a plugin, if you make it easy to install via npm we can post it on the documentation site for others to use too 👍
